pkg/resources/watcher_test.go

```diff
@@ -0,0 +1,40 @@
+package resources
+
+import (
+	"github.com/Mellanox/k8s-rdma-shared-dev-plugin/pkg/utils"
+	"os"
+	"syscall"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Server", func() {
+	Context("newFSWatcher", func() {
+		It("Watcher for existing Dir", func() {
+			fs := utils.FakeFilesystem{}
+			defer fs.Use()()
+			fsw, err := newFSWatcher(fs.RootDir)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(fsw).ToNot(BeNil())
+		})
+		It("Watcher for non-existing Dir", func() {
+			fsw, err := newFSWatcher("fake")
+			Expect(err).To(HaveOccurred())
+			Expect(fsw).To(BeNil())
+		})
+	})
+	Context("NewOSWatcher", func() {
```

Comment:
thats more of a functional test then a unit test.

ideally you should mock signla.Notify and make sure it was called with the correct signals.

Reply:
also what will happen if no signals where passed ?

according to signal.Notify() docs:

> Notify causes package signal to relay incoming signals to c. If no signals are provided, all incoming signals will be relayed to c. Otherwise, just the provided signals will.

is that whats expected from NewOsWatcher ?

Reply:
should be called NewSignalNotifier() imo but anyway.....

Reply:
Added mock for it, and changed the name to NewSignalNotifier.

Regarding the behavior, not sure if we need all the signals to be notified if signal.Notify() is called without provided signals, maybe we can change it in other PR if need.